apps/ai-content-generator/src/components/config/content-type-selection/ContentTypeSelection.tsx

```diff
@@ -0,0 +1,74 @@
+import { Dispatch, useState } from 'react';
+import { Checkbox, FormControl, Table } from '@contentful/f36-components';
+import { ContentTypeProps } from 'contentful-management';
+import { ContentTypeText } from '@components/config/configText';
+import { ContentTypeAction, ContentTypeReducer } from '@components/config/contentTypeReducer';
+import { styles } from './ContentTypeSelection.styles';
+
+interface Props {
+  allContentTypes: ContentTypeProps[];
+  selectedContentTypes: string[];
+  dispatch: Dispatch<ContentTypeReducer>;
+}
+
+const ContentTypeSelection = (props: Props) => {
+  const { allContentTypes, selectedContentTypes, dispatch } = props;
+  const [isSelectAllChecked, setIsSelectAllChecked] = useState(false);
+
+  const allContentTypeIds = allContentTypes.map((ct) => ct.sys.id);
```

Comment:
Could we just move this map inside of the code block on line 24? That way this doesn't get called on every rerender